apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -11,28 +12,78 @@ import { DelayedJob } from '@terramatch-microservices/database/entities';
 
 @Controller('jobs/v3/delayedJobs')
 export class DelayedJobsController {
+  @Get()
+  @ApiOperation({
+    operationId: 'listDelayedJobs',
+    description: 'Retrieve a list of all delayed jobs.',
+  })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Authentication failed.',
+  })
+  async getRunningJobs(
+    @Request() { authenticatedUserId }
+  ): Promise<JsonApiDocument> {
+    const runningJobs = await DelayedJob.findAll({
+      where: {
+        isCleared: false,
+        createdBy: authenticatedUserId
+      },
+      order: [['createdAt', 'DESC']],
+    });
+
+    const document = buildJsonApi();
+    runningJobs.forEach((job) => {
+      document.addData(job.uuid, new DelayedJobDto(job));
+    });
+    return document.serialize();
+  }
+
   @Get(':uuid')
   @ApiOperation({
     operationId: 'delayedJobsFind',
-    description: 'Get the current status and potentially payload or error from a delayed job.',
+    description:
+      'Get the current status and potentially payload or error from a delayed job.',
   })
   @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, {
     description: 'Authentication failed.',
   })
   @ApiException(() => NotFoundException, {
-    description: 'Job with that UUID not found.'
+    description: 'Job with that UUID not found.',
   })
+  // Note: Since jobs are very generic and we don't track which resources are related to a given
+  // job, there is no effective way to make a policy for jobs until we expand the service to
+  // include an owner ID on the job table.
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
-
     return buildJsonApi()
       .addData(pathUUID, new DelayedJobDto(job))
       .document.serialize();
   }
-}
+
+  @Patch('clear')
+  @ApiOperation({
+    operationId: 'clearNonPendingJobs',
+    description: 'Set isCleared to true for all jobs where status is not pending.',
+  })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Authentication failed.',
+  })
+  async clearNonPendingJobs(@Request() { authenticatedUserId }): Promise<{ message: string }> {
```

Comment:
I think this blanket "clear all" approach is likely to lead to bugs, and is not very RESTful. It would be much better for the FE to send the list of jobs it would like to clear out. This should be done either as a series of calls to a `@Patch(':uuid')` endpoint, or as a bulk operation. See the research service for  an example of a bulk patch operation. The payload in either case should be sent in JsonAPI format, correctly highlighting the resources by ID and the field(s) that should be updated.

Another possible approach to this problem would be to remove the `isCleared` flag, and do a `Delete()` endpoint instead. One logical problem with this endpoint as written is that it's possible to "clear" a job that hasn't been finished yet, which seems odd to me. Instead, we could do a delete endpoint that takes a set of job UUIDs a query params and deletes them. In that case (and maybe even if we keep this `isCleared` pattern), the endpoint should throw an error if the FE tries to delete / clear a job that is still pending.

Reply:
I change it to a bulk operation, considering the API format for the input.